apps/election-manager/src/components/ExportFinalResultsModal.tsx

```diff
@@ -0,0 +1,288 @@
+      const { inputFiles, outputFiles } = await client.getFiles()
+      const [electionDefinitionFile, cvrFile] = inputFiles
+      const resultsFile = outputFiles[0]
+
+      await client.setInputFile(
+        electionDefinitionFile.name,
+        new File(
+          [electionDefinition!.electionData],
+          electionDefinitionFile.name,
+          {
+            type: 'application/json',
+          }
+        )
+      )
+      await client.setInputFile(
+        cvrFile.name,
+        new File([CastVoteRecordsString], 'cvrs')
+      )
+      await client.process()
+
+      // download the result
+      const results = await client.getOutputFile(resultsFile.name)
+      if (!window.kiosk) {
+        fileDownload(results, defaultFilename, 'text/csv')
+      } else {
+        const usbPath = await getDevicePath()
+        const pathToFile = path.join(usbPath!, defaultFilename)
+        if (openFileDialog) {
+          const fileWriter = await window.kiosk.saveAs({
+            defaultPath: pathToFile,
+          })
+
+          if (!fileWriter) {
+            throw new Error('could not begin download; no file was chosen')
+          }
+
+          await fileWriter.write(await results.text())
+          setSavedFilename(fileWriter.filename)
+          await fileWriter.end()
+        } else {
+          await window.kiosk!.writeFile(pathToFile, await results.text())
+          setSavedFilename(defaultFilename)
+        }
+      }
+
+      // reset server files
+      await client.reset()
+      await new Promise((resolve) => setTimeout(resolve, 2000))
+      setCurrentState(ModalState.DONE)
+    } catch (error) {
+      setErrorMessage(`Failed to save results. ${error.message}`)
+      setCurrentState(ModalState.ERROR)
+    }
+  }
+
+  if (currentState === ModalState.ERROR) {
+    return (
+      <Modal
+        isOpen={isOpen}
+        content={
+          <Prose>
+            <h1>Saving Results Failed</h1>
+            <p>{errorMessage}</p>
+          </Prose>
+        }
+        onOverlayClick={onClose}
+        actions={<LinkButton onPress={onClose}>Close</LinkButton>}
+      />
+    )
+  }
+
+  if (currentState === ModalState.DONE) {
+    let actions = (
+      <React.Fragment>
+        <LinkButton onPress={onClose}>Close</LinkButton>
+        <USBControllerButton small={false} primary />
+      </React.Fragment>
+    )
+    if (usbDriveStatus === UsbDriveStatus.recentlyEjected) {
+      actions = <LinkButton onPress={onClose}>Close</LinkButton>
+    }
+    return (
+      <Modal
+        isOpen={isOpen}
+        content={
+          <Prose>
+            <h1>Results Saved</h1>
+            <p>
+              Results file succesfully saved{' '}
+              {savedFilename !== '' && (
+                <span>
+                  as <strong>{savedFilename}</strong>
+                </span>
+              )}{' '}
+              directly on the inserted USB drive. You may now eject the USB.
+            </p>
```

Comment:
- Move the main action we want them to take to the top of content.
- Insert "drive" at the end of the sentence.
- Fix typo in "successfully".

```suggestion
            <p>
              You may now eject the USB drive.
            </p>
            <p>
              Results file successfully saved{' '}
              {savedFilename !== '' && (
                <span>
                  as <strong>{savedFilename}</strong>
                </span>
              )}{' '}
              directly on the inserted USB drive.
            </p>
```